test/appium/support/testrail_report.py

```diff
@@ -0,0 +1,68 @@
+import json
+import requests
+import base64
+from os import environ
+from support.base_test_report import BaseTestReport
+
+
+class TestrailReport(BaseTestReport):
+
+    def __init__(self, sauce_username, sauce_access_key):
+        super(TestrailReport, self).__init__(sauce_username, sauce_access_key)
+
+        self.password = environ.get('TESTRAIL_PASS')
+        self.user = environ.get('TESTRAIL_USER')
+
+        self.run_id = None
+        self.suite_id = 42
+        self.project_id = 9
+
+        self.outcomes = {
+            'passed': 1,
+            'undefined_fail': 10}
+
+        self.headers = dict()
+        self.headers['Authorization'] = 'Basic %s' % str(
+            base64.b64encode(bytes('%s:%s' % (self.user, self.password), 'utf-8')), 'ascii').strip()
+        self.headers['Content-Type'] = 'application/json'
+
+        self.url = 'https://ethstatus.testrail.net/index.php?/api/v2/'
+
+    def get(self, method):
+        raw_response = requests.get(self.url + method, headers=self.headers).text
+        return json.loads(raw_response)
+
+    def post(self, method, data):
+        data = bytes(json.dumps(data), 'utf-8')
+        raw_response = requests.post(self.url + method, data=data, headers=self.headers).text
+        return json.loads(raw_response)
+
+    def get_suites(self, project_id):
+        return self.get('get_suites/%s' % project_id)
+
+    def get_tests(self, run_id):
+        return self.get('get_tests/%s' % run_id)
+
+    def get_milestones(self, project_id):
+        return self.get('get_milestones/%s' % project_id)
+
+    def add_run(self, run_name):
+        request_body = {'suite_id': self.suite_id,
+                        'name': run_name,
+                        'milestone_id': self.get_milestones(self.project_id)[-1]['id']}
```

Comment:
Maybe change `self.get_milestones(self.project_id)[-1]['id']` to `get_last_mailstone_id`